application/account-management/WebApp/src/shared/router/generate.tsx

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bun
+// @ts-
+
+import fs from "fs";
+import path from "path";
+import prettier from "prettier";
+import { RoutePage, RouteType, getRouteDetails } from "./routeDetails";
+import {
+  autoGeneratedBannerCode,
+  createBrowserRouterCode,
+  errorPageCode,
+  importCode,
+  layoutPageCode,
+  lazyLoadingPageCode,
+  normalPageCode,
+} from "./pageCode";
+
+const sourceRoot = fs.existsSync(path.join(process.cwd(), "src")) ? path.join(process.cwd(), "src") : process.cwd();
+const outputPath = path.join(sourceRoot, "lib", "router", "router.generated.ts");
+const appPath = path.join(sourceRoot, "app");
+const importPrefix = "@";
+
+if (!fs.existsSync(appPath)) {
+  throw new Error(`Could not find app directory at ${appPath}`);
+}
+
+type TemplateObject = Record<string, string>;
+
+function generateReactRouterCode(routeItem: RouteType): string {
+  if (routeItem.type === "page") {
+    if (routeItem.loadingPage) {
+      const landingRoutePage: RoutePage = { ...routeItem, page: routeItem.loadingPage };
+      return `{
+        index: true,
+        element: ${lazyLoadingPageCode(routeItem, landingRoutePage)},
+      }`;
+    }
+    return `{
+      index: true,
+      element: ${normalPageCode(routeItem)},
+    }`;
+  }
+  if (routeItem.type === "not-found") {
+    return `{
+      path: "*",
+      element: ${normalPageCode(routeItem)},
+    }`;
+  }
+
+  if (routeItem.type === "entry") {
+    const result: TemplateObject = {};
+
+    if (routeItem.layout != null) {
+      result.element = layoutPageCode(routeItem.layout);
+    }
+
+    if (routeItem.error != null) {
+      result.errorElement = errorPageCode(routeItem.error);
+    }
+
+    if (routeItem.children.length > 0) {
+      result.children = `[${routeItem.children.map((c) => generateReactRouterCode(c)).join(",\n")}]`;
+    }
+
+    return routeItem.aliases
+      .map((alias) =>
+        serializeTemplateObject({
+          path: `"${alias}"`,
+          ...result,
+        })
+      )
+      .join(",\n");
+  }
+
+  throw new Error(`Unhandled route type ${routeItem.type}`);
```

Comment:
In C#, we follow the convention where all exception messages end with a trailing period, making them read like complete sentences. Although there are no strict global conventions for log messages in C#, I've opted for consistency by also always adding a trailing period to these, as this allows for multi-sentence messages while maintaining consistency.

Regarding TypeScript, do we have a similar strict convention? It would be beneficial to adopt a uniform approach across languages.

I've noticed the same convention of adding a trailing period in `WebApp/src/shared/rspack/runtime.ts`. Additionally, in that file there's the use of an `Error:` prefix in the messages. Personally, I think this prefix is a bit redundant, I believe consistency is key here.

Reply:
Not sure if there's a convention for this in js - but we could make it so in PP that exceptions have to read and end as one sentence. The prefix `Error:` help filter output in the terminal - we could use a real logger with builtin filtering etc. (But not keen on adding more dependencies)

Reply:
Agree that "Error:" is redundant in exceptions :)